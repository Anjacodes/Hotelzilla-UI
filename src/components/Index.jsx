import React from 'react';
import RoomItem from './RoomItem';
import { useSelector } from 'react-redux';

function Index() {
  const rooms = [
    {
      id: 1,
      name: 'Room 1',
      image: 'http://via.placeholder.com/640x360',
      description: 'This is room1 description',
    },
    {
      id: 2,
      name: 'Room 2',
      image: 'http://via.placeholder.com/640x360',
      description: 'This is room2 description',
    },
    {
      id: 3,
      name: 'Room 3',
      image: 'http://via.placeholder.com/640x360',
      description: 'This is room3 description',
    },
    {
      id: 4,
      name: 'Room 4',
      image: 'http://via.placeholder.com/640x360',
      description: 'This is room4 description',
    },
  ];

  const role = useSelector((state) => state.login.role);

  return (
    <>
      <div className="flex flex-col">
        <div className="flex justify-end p-5">
          {role === 'Admin' && (
            <button className="bg-lime-400 px-3 py-2 text-white">
              Add Room
            </button>
          )}
        </div>
        <div className="grid grid-cols-3 grid-rows-3 gap-6 p-5">
          {rooms.map((room) => (
            <RoomItem key={room.id} room={room} />
          ))}
        </div>
      </div>
    </>
  );
}

export default Index;
