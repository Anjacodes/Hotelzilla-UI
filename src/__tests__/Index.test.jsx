/**
 * @vitest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { it, describe, expect } from 'vitest';
import store from '../redux/configStore';
import Index from '../components/Index';

describe('Index', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Index />
        </BrowserRouter>
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
