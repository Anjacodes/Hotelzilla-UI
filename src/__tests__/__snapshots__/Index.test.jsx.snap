// Vitest Snapshot v1

exports[`Hotel Item > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center"
      >
        <div
          class="flex h-screen items-center justify-center"
        >
          <i
            class="fa-solid fa-caret-left absolute left-0 h-[50px] rounded-r-full bg-gray-200 py-4 px-6 text-slate-50 md:left-[20%] hover:not"
          />
          <div
            class="grid w-10/12 grid-cols-1 gap-3 p-5 md:grid-cols-3 md:grid-rows-1 sm:"
          />
          <i
            class="fa-solid fa-caret-right absolute right-0 h-[50px] rounded-l-full bg-lime-400 py-4 px-6 text-slate-50 hover:not"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center"
    >
      <div
        class="flex h-screen items-center justify-center"
      >
        <i
          class="fa-solid fa-caret-left absolute left-0 h-[50px] rounded-r-full bg-gray-200 py-4 px-6 text-slate-50 md:left-[20%] hover:not"
        />
        <div
          class="grid w-10/12 grid-cols-1 gap-3 p-5 md:grid-cols-3 md:grid-rows-1 sm:"
        />
        <i
          class="fa-solid fa-caret-right absolute right-0 h-[50px] rounded-l-full bg-lime-400 py-4 px-6 text-slate-50 hover:not"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Index > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center"
      >
        <div
          class="flex h-screen items-center justify-center"
        >
          <i
            class="fa-solid fa-caret-left absolute left-0 h-[50px] rounded-r-full bg-gray-200 py-4 px-6 text-slate-50 md:left-[20%] hover:not"
          />
          <div
            class="grid w-10/12 grid-cols-1 gap-3 p-5 md:grid-cols-3 md:grid-rows-1 sm:"
          />
          <i
            class="fa-solid fa-caret-right absolute right-0 h-[50px] rounded-l-full bg-lime-400 py-4 px-6 text-slate-50 hover:not"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center"
    >
      <div
        class="flex h-screen items-center justify-center"
      >
        <i
          class="fa-solid fa-caret-left absolute left-0 h-[50px] rounded-r-full bg-gray-200 py-4 px-6 text-slate-50 md:left-[20%] hover:not"
        />
        <div
          class="grid w-10/12 grid-cols-1 gap-3 p-5 md:grid-cols-3 md:grid-rows-1 sm:"
        />
        <i
          class="fa-solid fa-caret-right absolute right-0 h-[50px] rounded-l-full bg-lime-400 py-4 px-6 text-slate-50 hover:not"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
